### PR TITLE
Patterns

### LINKED ISSUE #1: Service-registry
Implement service-registry pattern using Spring Cloud and/or K8s

### LINKED ISSUE #2: Api-gateway
Implement api-gateway pattern using Spring Cloud and/or K8s

### LINKED ISSUE #4: repository is monolithic
It should be split into repo/per service.
Read more: https://www.yegor256.com/2018/09/05/monolithic-repositories.html

### PR DESCRIPTION
**Pull request by dukris**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+- Book microservice
+- Author microservice
+- Spring Cloud API Gateway
+- Netflix Eureka service registry
+
+## Glossary
+1. Author
+2. Book
+
+## Endpoints
```
Comment (h1alexbel) [inline README.md]: Use swagger instead

File: book-service/src/main/resources/application.yml [hunk 2]
```diff
@@ -1,11 +1,11 @@
 server:
-  port: 8081
+  port: 8082
 
 spring:
   config:
     import: .env[.properties]
   application:
-    name: book
+    name: book-service
```
Comment (h1alexbel) [inline book-service/src/main/resources/application.yml:8]: not required. Better to name services without ```-service```

**Review by h1alexbel — CHANGES_REQUESTED** [review]
see #4

**Comment by dukris** [comment]
Please, see: [AuthorMicroservice](https://github.com/hizmailovich/AuthorMicroservice), [BookMicroservice](https://github.com/hizmailovich/BookMicroservice), [EurekaServer](https://github.com/hizmailovich/EurekaServer), [ApiGateway](https://github.com/hizmailovich/ApiGateway).

File: author-service/src/main/resources/application.yml [hunk 3]
```diff
@@ -15,3 +15,8 @@ spring:
     hibernate:
       ddl-auto: update
     database: postgresql
+
+eureka:
+  client:
+    service-url:
+      default-zone: http://localhost:8761/eureka
```
Comment (h1alexbel) [inline author-service/src/main/resources/application.yml]: move into ${HOST}

**Review by h1alexbel — CHANGES_REQUESTED** [review]

File: author-service/src/main/resources/application.yml (hunk 3 above)
Comment (dukris) [reply author-service/src/main/resources/application.yml]: This bug is fixed, see: [AuthorMicroservice](https://github.com/hizmailovich/AuthorMicroservice) and [BookMicroservice](https://github.com/hizmailovich/BookMicroservice).

**Pull request merged**